--- a/chrome/chrome_browser.gypi	2015-10-17 16:23:34.168610541 +0200
+++ b/chrome/chrome_browser.gypi	2015-10-17 16:24:35.652774539 +0200
@@ -3559,7 +3559,7 @@
             '../build/linux/system.gyp:gnome_keyring',
           ],
         }],
-        ['OS=="linux" and chromeos!=1', {
+        ['use_libsecret==1', {
           'sources': [ '<@(chrome_browser_libsecret_sources)' ],
           'defines': [
             'USE_LIBSECRET',
--- a/chrome/browser/BUILD.gn	2015-10-17 18:53:01.330828095 +0200
+++ b/chrome/browser/BUILD.gn	2015-10-17 18:53:24.322850924 +0200
@@ -515,7 +515,6 @@
     sources += rebase_path(gypi_values.chrome_browser_libsecret_sources,
                            ".",
                            "//chrome")
-    defines += [ "USE_LIBSECRET" ]
   }
   if (use_aura) {
     sources +=
--- a/build/common.gypi	2015-10-17 18:00:08.769821670 +0200
+++ b/build/common.gypi	2015-10-17 19:03:24.983520481 +0200
@@ -753,9 +753,12 @@
         # Flags to use glib.
         ['OS=="win" or OS=="mac" or OS=="ios" or OS=="android" or use_ozone==1', {
           'use_glib%': 0,
+	  'use_libsecret%': 0,
         }, {
           'use_glib%': 1,
+	  'use_libsecret%': 1,
         }],
+	
 
         # Flags to use pango and cairo.
         ['OS=="win" or OS=="mac" or OS=="ios" or OS=="android" or embedded==1', {
@@ -1114,6 +1117,7 @@
     'os_posix%': '<(os_posix)',
     'use_dbus%': '<(use_dbus)',
     'use_glib%': '<(use_glib)',
+    'use_libsecret%': '<(use_libsecret)',
     'use_pango%': '<(use_pango)',
     'use_cairo%': '<(use_cairo)',
     'use_ozone%': '<(use_ozone)',
