--- ./third_party/skia/src/gpu/gl/egl/SkCreatePlatformGLContext_egl.cpp.orig	2015-10-02 16:53:03.655176400 +0200
+++ ./third_party/skia/src/gpu/gl/egl/SkCreatePlatformGLContext_egl.cpp	2015-10-02 18:19:05.471760489 +0200
@@ -6,7 +6,7 @@
  * found in the LICENSE file.
  */
 #include "gl/SkGLContext.h"
-
+#include "ui/gfx/x/x11_types.h"
 #include <GLES2/gl2.h>
 #include <EGL/egl.h>
 
@@ -70,7 +70,7 @@
     SkASSERT(forcedGpuAPI == kNone_GrGLStandard || kAPIs[api].fStandard == forcedGpuAPI);
 
     for (; NULL == fGL.get() && api < apiLimit; ++api) {
-        fDisplay = eglGetDisplay(EGL_DEFAULT_DISPLAY);
+        fDisplay = eglGetDisplay(gfx::GetXDisplay());
 
         EGLint majorVersion;
         EGLint minorVersion;
